src/components/KetcherWindow.vue

```diff
@@ -18,8 +46,63 @@ export default {
   name: "KetcherWindow",
   data() {
     return {
-      ketcherURL: process.env.VUE_APP_KETCHER_URL
+      ketcherURL: process.env.VUE_APP_KETCHER_URL,
+      molfile: ""
     };
+  },
+  methods: {
+    importMolfile: function() {
+      document
+        .getElementById("ketcher")
+        .contentWindow.postMessage(
+          { type: "importMolfile", molfile: this.molfile },
+          "*"
+        );
+    },
+    loadMolfile: function() {
+      document.getElementById("ketcher").contentWindow.postMessage(
+        {
+          type: "importMolfile",
+          molfile: this.$store.state.compound.molfile
+        },
+        "*"
+      );
+    },
+    exportMolfile: function() {
+      document
+        .getElementById("ketcher")
+        .contentWindow.postMessage({ type: "exportMolfile" }, "*");
+    }
+  },
+  computed: {
+    filio: function() {
+      return this.$store.state.compound.molfile;
+    }
+  },
+  watch: {
+    filio: function() {
+      this.loadMolfile();
+    }
+  },
+  mounted() {
+    let self = this;
+    window.addEventListener(
+      "message",
+      function(event) {
+        if (event.data.type == "returnMolfile") {
+          self.molfile = event.data.molfile;
+          this.molfile = this.$store.state.compound.molfile;
+        }
+      },
+      false
+    );
+    if (this.$store.state.compound.molfile !== "") {
+      const load = this.loadMolfile; // lexical this, goes away in setTimeout!
+      setTimeout(function() {
+        load();
+      }, 300); // there is a bit of lag to get the .ketcher element to load, thus, the timeout.
```

Comment:
This is going to be fickle. The 300ms delay will work for local machines while testing, but since this is an iframe, I expect longer latency in production.

A more robust solution is to have the iframe emit an event to its parent (our Vue app) that it is ready.

Reply:
adding an EventListener to the "load" event of the iframe will work for this.
```
var iFrame = document.getElementById("ketcher");
iFrame.addEventListener("load", function() {
  load();
});
```